core/templates/services/UpgradedServices.ts

```diff
@@ -343,6 +345,8 @@ export class UpgradedServices {
     upgradedServices['ExplorationFeaturesService'] =
       new ExplorationFeaturesService();
     upgradedServices['ExpressionParserService'] = new ExpressionParserService();
+    upgradedServices['ExpressionSyntaxTreeService'] =
+      new ExpressionSyntaxTreeService(new ExpressionParserService());
```

Comment:
Do not create a second `ExpressionParserService`. You should pass the one already constructed (on line 347) instead. Also, please move this into the next group of services.

```typescript
upgradedServices['ExpressionSyntaxTreeService'] =
  new ExpressionSyntaxTreeService(upgradedServices['ExpressionParserService']);
```

Reply:
> Also, please move this into the next group of services.

@brianrodri Don't get it, the next group of services?

Reply:
Right, this should be under the `Topological level: 1` group ([link](https://github.com/oppia/oppia/blob/develop/core/templates/services/UpgradedServices.ts#L421)), because it depends on something from the `Topological level: 0` group.

For background, `Topological level: 0` means you take zero arguments (like `ExpressionParserService`).

Level 1 means you take an argument from something in level 0.
Level 2 means you take an argument from something in level 1.
... and so on.

Reply:
@brianrodri Ah I see, interesting! 👍